### PR TITLE
feat: add warp ds eslint config

### PR DESCRIPTION
**Pull request by felicia-haggqvist**
Fixes Jira issue: [WARP-520](https://nmp-jira.atlassian.net/browse/WARP-520)

- Add eslint configuration and workflow
- lint all code using `pnpm lint`

### CODE REVIEW

File: vite.config.js [hunk 1]
```diff
@@ -22,23 +22,7 @@ export default function config(env) {
...
-          "lingui.config.ts",
-          "@types",
-          "tests/",
-          "packages/**/src/props.*",
-          "packages/**/src/index.*",
-          "packages/index.ts",
-          "build.js",
-          "esbuild.js",
-        ],
+        exclude: ['**.json', 'dev/**', 'storybook-static/**', '.storybook/**', 'packages/**/stories', 'packages/**/src/locales', 'esbuild.mjs', 'lingui.config.ts', '@types', 'tests/', 'packages/**/src/props.*', 'packages/**/src/index.*', 'packages/index.ts', 'build.js', 'esbuild.js'],
```
Comment (BalbinaK) [inline vite.config.js:25]: I can't wrap my head around why this wrapped content was formated to be one line, while in other repos that didn't happen. We had the same setup e.g. in @warp-ds/vue, which extends @warp-ds/eslint-config. I see that in our Warp config we specified `printWidth` for prettier to 999 ([code](https://github.com/warp-ds/eslint-config/blob/master/index.js#L52C4-L58C7)), which sounds like the setting responsible for this, but it only seems to apply in this repo. I'm also wondering, is `printWidth: 999` rule for prettier something we would like to have? @warp-ds/warp-core-team

Comment (felicia-haggqvist) [reply vite.config.js:25]: Yeah I don't know why it didn't format the same way as for vue 🙈 😕

Comment (imprashast) [reply vite.config.js:25]: It shouldn't exceed 140 `"printWidth": 140,`
This would work on most retina display macbooks without issues

Comment (imprashast) [reply vite.config.js:25]: It also might be because of `"plugin:prettier/recommended"` in the eslintrc 🤔

Comment (felicia-haggqvist) [reply vite.config.js:25]: No it did that formatting before I added `"plugin:prettier/recommended"` in the eslintrc

Comment (felicia-haggqvist) [reply vite.config.js:25]: Although it looks like it had atleast formatted the `exclude` the same way in the `vite.config.js`file for vue. 🤔 See [here](https://github.com/warp-ds/vue/pull/152/files#diff-58e6f63d87181b1c6a8cb6e5f1691df04aa32854456efcd52ca71c8541375d26R27)

Comment (BalbinaK) [reply vite.config.js:25]: Isn't it the `"printWidth": 999` from [@warp-ds/eslint-config](https://github.com/warp-ds/eslint-config/blob/master/index.js#L52C4-L58C7) that is kicking in here (and then it didn't in the other repos?)? When I added `printWidth: 150` as a separate rule in `.eslintrc` here, it worked fine. 999 characters is an eternity in the world of an IDE, so I'm not sure why it was set to that in the first place 😅

Comment (felicia-haggqvist) [reply vite.config.js:25]: But it reacted the same way in vue as in react (see my previous comment). So it did kick in. Talked to @imprashast and we talked about that I should update the @warp-ds/eslint-config to have `printWidth: 140` :)

File: .github/workflows/lint.pr.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: lts/*
+    - name: Install pnpm and dependencies
+      uses: pnpm/action-setup@v2
+      with:
+        version: 8
```
Comment (BalbinaK) [inline .github/workflows/lint.pr.yaml:18]:
```suggestion
        version: 9
```

**Review by BalbinaK — COMMENTED** [review]
Just one comment and we should be good to go!

**Review by BalbinaK — APPROVED** [review]

**Pull request merged**